### PR TITLE
Changes to testing file matching patterns #6321

### PR DESCRIPTION
**Pull request by tmeasday**

### CODE REVIEW

File: tools/isobuild/package-source.js [hunk 1]
```diff
@@ -1386,10 +1391,14 @@ _.extend(PackageSource.prototype, {
     // Ignore the usual ignorable files.
     sourceReadOptions.exclude.push(...ignoreFiles);
 
-    // Unless we're running tests, ignore source files with name
-    // "*test*.*", "*tests*.*", "test.*", "tests.*"
+    // Unless we're running tests, ignore all test filenames and if we are, ignore the
+    // type of file we *aren't* running
     if (!global.testCommandMetadata) {
-      Array.prototype.push.apply(sourceReadOptions, TEST_FILENAME_REGEXPS);
+      Array.prototype.push.apply(sourceReadOptions.exclude, TEST_FILENAME_REGEXPS);
```
Comment (avital) [inline tools/isobuild/package-source.js]: I think you also need to push `UNIT_TEST_FILENAME_REGEXPS` and `INTEGRATION_TEST_FILENAME_REGEXPS` here.

But really, I'd change it a bit and have `TEST_FILENAME_REGEXPS` include both of those already where it's defined (using the ES6 array spread operator), and actually not change the code here.

And then I'd rename `TEST_FILENAME_REGEXPS` to `ALL_TEST_FILENAME_REGEXPS`. That makes the logical relationship between these three arrays match their names.

File: tools/isobuild/app-test-files.js [hunk 2]
```diff
@@ -8,6 +8,18 @@ export const TEST_FILENAME_REGEXPS = [
   /^tests?./
```
Comment (avital) [inline tools/isobuild/app-test-files.js]: At this point we might consider just getting rid of the `tests?.*` form... Not only does it stick out in it's difference from the unit test and integration test filenames, and makes it harder to exactly explain how this feature works -- it also makes colocating tests not as great. It's much better to see `foo.js` and `foo.test.js` right next to each other in your editor than a bunch of files and then a bunch of `test_*` files where you can't even easily see which files have tests.

What do you think?

**Comment by tmeasday** [comment]
@avital please review again

**Comment by avital** [comment]
Looks great! Worth verified at least once that `test-packages` still works (I'm pretty sure it does but the code changed is on that code path as well)

**Pull request merged**